### PR TITLE
Added getScriptsByCluster API call with required clusterId

### PR DESCRIPTION
**Pull request by thavlice**
Added clusterId parameter to  API as a change required for fixing `backplane script list` command.

### CODE REVIEW

File: openapi/openapi.yaml
Comment (wanghaoran1988) [inline openapi/openapi.yaml]: Will this breaks the existing cli functionality?

Comment (thavlice) [reply openapi/openapi.yaml]: As I believe, calls using`/backplane/script` verb dont not work in the CLI as intended at the moment, so changing it wouldn't matter, however I will double-check test more with the full pipeline and use cases in the CLI before merging here and further.

Alternative would be to add the verb without deleting it, will report when tested enough. 

As of now, I'll double check the process and provide more info.(Do not merge yet)

Comment (thavlice) [reply openapi/openapi.yaml]: I've looked into it and believe that the "migration" steps should be taken.
Because of that, I'm adding a new API instead of changing the current one as initially described in the [ticket](https://issues.redhat.com/browse/OSD-12531)

**Comment by thavlice** [comment]
@wanghaoran1988 After implementing the change, I have trouble figuring out what is the cause of failed test for the pipeline, since the change should not affect it. Any ideas?

File: openapi/openapi.yaml
Comment (Nikokolas3270) [reply openapi/openapi.yaml]: @thavlice : what is the motivation behind implementing a migration? 
As you said command `ocm backplane script list` is already broken; so there is no danger in changing the existing `getScripts` entry.

Comment (thavlice) [reply openapi/openapi.yaml]: @Nikokolas3270 I believe there was a mixup between the live version and the version tested. As of now, there are 2 pairs of API + CLI repositories.

- backplane API on GitHub: with OpenAPI definition
- backplane API on GitLab: with API handling
- backplane CLI on GitHub: with (fully) working CLI and maintained tests
- backplane CLI on GitLab: the CLI we initially tested against with the result not working, however as confirmed by @supreeth7, this repository is not maintained anymore and should not be used

In addition, the API definition was moved from API GitLab to Github, mixing up things more.

Therefore I did the testing on the correct "live" CLI version from GitHub and the command works just fine.

We are able to perform the migration as initially intended since it was assuming working CLI, which is the case now.

**Comment by thavlice** [comment]
The changes must be made so that the right BackplaneApi.go is generated correctly, which was not checked and caused the failing test.

File: openapi/openapi.yaml
Comment (bmeng) [inline openapi/openapi.yaml:216]: Should the summary here reflect that the call is to the given cluster?

File: openapi/openapi.yaml
Comment (bmeng) [inline openapi/openapi.yaml]: The name here looks a bit weird... I am not quite familiar with the OpenAPI convention, but it should be better if we name this as something like `getScriptsByCluster`
Correct me if I am understanding wrong...

Comment (supreeth7) [reply openapi/openapi.yaml]: I agree, we should ideally not have numerics in the operationId, how about we call it something like `getClusterScripts` or `getScriptsByCluster` as Bo suggested above.

**Comment by supreeth7** [comment]
@thavlice question:
I see you were looking to deprecate `/backplane/script`, can you please verify it won't have any breaking changes?
IIRC this _was_ used by our CLI lately.

File: openapi/openapi.yaml
Comment (thavlice) [reply openapi/openapi.yaml]: I agree, the name is not ideal, I couldn't come up with a better one at the time.
I can change it to `getScriptsByCluster`, seems like a good fitting name.

**Comment by thavlice** [comment]
> @thavlice question: I see you were looking to deprecate `/backplane/script`, can you please verify it won't have any breaking changes? IIRC this _was_ used by our CLI lately.

@supreeth7  At first I wanted to change the the API completely, because I was assuming the command was not working (but that was just a confusion of repositories I tested it against, as explained  [above](https://github.com/openshift/backplane-api/pull/7#discussion_r1162631030))

Now the plan is the "migration" approach (as more described [here,](https://issues.redhat.com/browse/OSD-12531) meaning we add a new verb and plug it to the handler in the other repository and follow up with changes in CLI to use the new verb `/backplane/script/{clusterId}` instead of the old one (so it won't be needed anymore).

The initial assumption was that the change was the only use of this verb in CLI, but good point. I will investigate more and confirm if we need to keep the original `/backplane/script`.

File: openapi/openapi.yaml
Comment (thavlice) [reply openapi/openapi.yaml:216]: @bmeng The semantics of  `/backplane/script/{cluster}` are the same as the `/backplane/script`. 
The `/backplane/script` already takes cluster into consideration, but it is not mentioned in the summary.
To compare with other commands, for example, `/backplane/script/{clusterId}/job`has `clusterId` in path, but the summary:
`get:
      operationId: getAllJobs
      summary: Get all job runs in the namespace the user has access to`
doesn't mention cluster as well.

So I believe the summary should stay the same as the semantics didn't change.

**Comment by thavlice** [comment]
> @thavlice question: I see you were looking to deprecate `/backplane/script`, can you please verify it won't have any breaking changes? IIRC this _was_ used by our CLI lately.

AFAIK the verb is used only by two commands and their corresponding files in backplane-CLI, list - `listScripts.go` and describe -`describeScript.go`.

I will plug the corresponding changes in later CLI PRs to both of these commands, so nothing else should be affected.

File: openapi/openapi.yaml
Comment (Nikokolas3270) [reply openapi/openapi.yaml]: Sorry I went on holidays for quite some time.
Indeed taking a look back at the following commit which introduced the issue:
https://gitlab.cee.redhat.com/service/backplane-api/-/commit/7717296c67054a8b5ee8f1661709e8949870f433
I looks like the `ocm backplane script list` is only broken for users with a backplane role not set for all clusters.
Typically MT SRE users are granted backplane access on a per cluster id or organisation id basis and this command is broken for them.
Users like us (SREP) are granted backplane access to all clusters... and command `ocm backplane script list` has not been broken for us; so with that in mind @thavlice: you are right, we should implement this change with a 2 steps migration to not break our ability to run this this command.

File: openapi/openapi.yaml
Comment (Nikokolas3270) [reply openapi/openapi.yaml:216]: `/backplane/script` does NOT take the cluster into consideration as the cluster id is not conveyed on this path.
That said, `/backplane/script/{cluster}` is gonna replace `/backplane/script`, so they could keep the same description indeed... especially if other paths conveying the cluster id do not mention they are specific to a given cluster.

File: openapi/openapi.yaml
Comment (Nikokolas3270) [reply openapi/openapi.yaml]: Other GET paths like `/backplane/cloud/credentials/{clusterId}` (or `/backplane/cloud/console/{clusterId}`) do not have the `operationId` suffixed with `ByCluster`.

If we don't like the `2` suffixing, what about:
- Renaming `getScripts` to `getScriptsOld`
- Creating the new path with the `operationId` set to `getScripts`?

Then once, the migration is finished, a PR would remove the old path with `operationId` set to `getScriptsOld`.

Remark that renaming an `operationId` is acceptable because:
- This id is not used in the server code.
- This id is used in the CLI code... but the use of the API is versioned:  
https://github.com/openshift/backplane-cli/blob/main/go.mod#L18. 
Also we have to modify CLI code anyway (to add the code for the new path); with that in mind we could sed the code to replace `GetScripts` with `GetScriptsOld` prior adding the code for the new path in the following locations:
https://github.com/search?q=repo%3Aopenshift%2Fbackplane-cli+getScripts&type=code

**Comment by Nikokolas3270** [comment]
> > @thavlice question: I see you were looking to deprecate `/backplane/script`, can you please verify it won't have any breaking changes? IIRC this _was_ used by our CLI lately.
> 
> @supreeth7 At first I wanted to change the the API completely, because I was assuming the command was not working (but that was just a confusion of repositories I tested it against, as explained [above](https://github.com/openshift/backplane-api/pull/7#discussion_r1162631030))
> 
> Now the plan is the "migration" approach (as more described [here,](https://issues.redhat.com/browse/OSD-12531) meaning we add a new verb and plug it to the handler in the other repository and follow up with changes in CLI to use the new verb `/backplane/script/{clusterId}` instead of the old one (so it won't be needed anymore).
> 
> The initial assumption was that the change was the only use of this verb in CLI, but good point. I will investigate more and confirm if we need to keep the original `/backplane/script`.

As explained above, the `ocm backplane script list` command is only working for SREP users or for users for which the backplane permission is granted to all clusters in ACM.
In that regards, the `/backplane/script` is NOT working as it does not work for all users; so I don't think there is point to keep it once `/backplane/script/{clusterId}` is in place and ALL users have migrated to a new version of the CLI using it.
The only reason I could see to keep it would be if this path was accessed outside any CLI call in which case we may wonder who/what is doing that.. and if there is a good reason doing so, then they will have to migrate to the new path as well.
I think it would be a really bad idea to keep both paths... this generates confusion on the long run not to mention probable code duplication :(.

**Comment by supreeth7** [comment]
Let's slowly phase out `/backplane/script` once this new API endpoint has been proven to work.

Thanks @Nikokolas3270 @thavlice, I'll merge this on Monday.

**Review by supreeth7 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
